src/eureka/S3_data_reduction/optspex.py

```diff
@@ -479,6 +480,74 @@ def profile_gauss(subdata, mask, threshold=10, guess=None, isplots=0):
     return profile
 
 
+def clean_median_flux(data, meta, log, median_thresh=5):
+    """Computes a median flux frame that is free of bad pixels.
+
+    Parameters
+    ----------
+    data : Xarray Dataset
+        The Dataset object.
+    meta : eureka.lib.readECF.MetaClass
+        The metadata object.
+    log : logedit.Logedit
+        The current log.
+    median_thresh : int; optional
+        Sigma threshold when flagging outliers in median frame
+
+    Returns
+    -------
+    data : Xarray Dataset
+        The updated Dataset object.
+
+    Notes
+    -----
+    History:
+
+    - 2022-08-03, KBS
+        Inital version
+    """
+    log.writelog('  Computing clean median frame...', mute=(not meta.verbose))
+
+    # Create mask using all data quality flags
+    mask = np.copy(data['mask'].values)
+    mask[np.where(data.dq > 0)] = 1
```

Comment:
This unmasks all pixels where the dq array is not good (or at least non-zero). I doubt that was your intended behaviour

Reply:
Aha, fixing this to be `= 0`, my issue with the Fig 3401 goes away (and at least right now I no longer have an issue with setting `vmax = vmin + 2000` although it does feel arbitrary)

Reply:
Here's my updated figure FYI

![fig3401_MedianFrame](https://user-images.githubusercontent.com/24514122/183507478-8bf58f11-a6ac-473d-ba2c-cbecf33f3c35.png)

Reply:
Relatedly, having masks = 0 where they're bad is quite confusing. We likely want to re-evaluate that in the future